BridgeSDK/Networking/SBBNetworkManager.m

```diff
@@ -417,7 +417,15 @@ - (NSURLSessionDataTask *) doDataTask: (NSString*) method
   NSMutableURLRequest *request = [self requestWithMethod:method URLString:URLString headers:headers parameters:parameters error:nil];
     __block NSURLSessionDataTask *task = [self.mainSession dataTaskWithRequest:request completionHandler:^(NSData *data, NSURLResponse *response, NSError *error) {
         NSError * httpError = [NSError generateSBBErrorForStatusCode:((NSHTTPURLResponse*)response).statusCode data:data];
-        NSDictionary * responseObject = [NSJSONSerialization JSONObjectWithData:data options:0 error:NULL];
+        id responseObject = nil;
+        if (data.length) {
+            responseObject = [NSJSONSerialization JSONObjectWithData:data options:0 error:NULL];
+            if (!responseObject) {
+                // maybe it was an html error page
+                responseObject = [[NSString alloc] initWithData:data encoding:NSUTF8StringEncoding];
```

Comment:
So, there's the situation where the iOS network call returns an error, which means there was a network error; and then if the request goes through and gets any kind of actual response  (including an http error response) it's up to the calling code to detect and handle that. I do this outside of the code block that checks for and handles errors so that the decoded response can be passed through to both "good" and "bad" paths. There's nothing here that will prevent proper error handling down the road, it just allows for that error handling code to log whatever might have come back, which in some odd cases I've seen to be raw html rather than json even when it hits one of our servers.